### PR TITLE
YR/Change last_run from int-to string/XSUP-32511

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
relates: https://jira-dc.paloaltonetworks.com/browse/XSUP-32511

## Description
Changing the way we save the UID in the last_run - from int to string.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by JasBeilin — APPROVED** [review]

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2_test.py [hunk 1]
```diff
@@ -597,3 +597,33 @@ def test_replace_spaces_in_credentials(input_credentials, output_credentials):
...
+        - A mock client and last run
+    When:  
+        - Fetching incidents
+    Then:
+        - Ensure that the "last_uid" received from the 'last_run' of previous cycles is converted to an integer.
+        Also, verify that the 'last_uid' to be written in the 'last_run' for the next cycle is a string.
+    """
+    from MailListenerV2 import fetch_incidents
+    mocker.patch('MailListenerV2.Email.convert_to_incident', return_value={})
+    fetch_mail_mocker = mocker.patch('MailListenerV2.fetch_mails', return_value=([mock_email()], [mock_email()], 5))
```
Comment (BEAdi) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2_test.py:614]: Maybe add a case where it a str and stays this way

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 2]
```diff
@@ -272,7 +272,7 @@ def fetch_incidents(client: IMAPClient,
         demisto.debug(f"no last_run, using {time_to_fetch_from=}")
 
     # Otherwise use the mail UID
-    uid_to_fetch_from = last_run.get('last_uid', 0)
+    uid_to_fetch_from = int(last_run.get('last_uid', 0))
```
Comment (BEAdi) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:275]: Consider adding logs, and using arg_to_number for better errors if they are raised.

**Review by BEAdi — APPROVED** [review]
Great job

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2_test.py (hunk 1 above)
Comment (RosenbergYehuda) [reply Packs/MailListener/Integrations/MailListenerV2/MailListenerV2_test.py:614]: will do

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py (hunk 2 above)
Comment (RosenbergYehuda) [reply Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:275]: There are enough logs from my perspective.

Changed to arg_to_number.

**Comment by ShirleyDenkberg** [comment]
@JasBeilin @BEAdi Doc review completed.  Looks OK.

**Pull request merged**